adapter_cyw43439.go

```diff
@@ -0,0 +1,119 @@
+//go:build cyw43439
+
+package bluetooth
+
+import (
+	"machine"
+
+	"log/slog"
+	"time"
+
+	"github.com/soypat/cyw43439"
+)
+
+const maxConnections = 1
+
+// Adapter represents a SPI connection to the HCI controller on an attached CYW4349 module.
+type Adapter struct {
+	hciAdapter
+}
+
+// DefaultAdapter is the default adapter on the current system.
+//
+// Make sure to call Enable() before using it to initialize the adapter.
+var DefaultAdapter = &Adapter{
+	hciAdapter: hciAdapter{
+		isDefault: true,
+		connectHandler: func(device Device, connected bool) {
+			return
+		},
+		connectedDevices: make([]Device, 0, maxConnections),
+	},
+}
+
+// Enable configures the BLE stack. It must be called before any
+// Bluetooth-related calls (unless otherwise indicated).
+func (a *Adapter) Enable() error {
+	time.Sleep(2 * time.Second)
+	if debug {
+		println("Initializing CYW43439 device")
+	}
+
+	dev := cyw43439.NewPicoWDevice()
+	cfg := cyw43439.DefaultBluetoothConfig()
+	if debug {
+		cfg.Logger = slog.New(slog.NewTextHandler(machine.USBCDC, &slog.HandlerOptions{
+			Level: slog.LevelDebug - 2,
+		}))
+	}
+
+	err := dev.Init(cfg)
+	if err != nil {
+		if debug {
+			println("Error initializing CYW43439 device", err.Error())
+		}
+		return err
+	}
+
+	transport := &hciSPI{dev: dev}
+
+	a.hci, a.att = newBLEStack(transport)
+	if debug {
+		println("Enabling CYW43439 device")
+	}
+
+	a.enable()
+
+	time.Sleep(1 * time.Second)
```

Comment:
Here as well, does this BLE stack require such long delays?

Reply:
Removing.